### PR TITLE
Set Upper Memory Limit for Go Tests

### PR DESCRIPTION
**Pull request by christoph-zededa**
Set an upper memory limit to know when we merge something that increases memory usage by a lot.
If this happens `test-profiling` can be used to look where the memory is going to.

### CODE REVIEW

**Comment by christoph-zededa** [comment]
> Do we hit the limit now?

No

**Comment by rouming** [comment]
> > Do we hit the limit now?
> 
> No

and if we set 400? :)

**Comment by christoph-zededa** [comment]
> > > Do we hit the limit now?
> > 
> > 
> > No
> 
> and if we set 400? :)

It sometimes fails with 600 on my laptop.

File: Makefile [hunk 1]
```diff
@@ -397,6 +397,9 @@ test: $(LINUXKIT) test-images-patches | $(DIST)
 	cp pkg/pillar/results.xml $(DIST)/
 	$(QUIET): $@: Succeeded
 
+test-profiling:
+	make -C pkg/pillar test-profiling
+
```
Comment (eriknordmark) [inline Makefile:402]: Should this be documented like the other Makefile targets (in "make all" output)?

Comment (christoph-zededa) [reply Makefile:402]: Yes, I added it to `make help`

**Review by eriknordmark — APPROVED** [review]
LGTM

**Pull request merged**